train.py

```diff
@@ -50,47 +55,118 @@ def write_info_file():
-        noisy = add_noise(data, args.sigma).copy()
-        np.savetxt(args.outf+'/samples/noisy' + str(image) + '.txt', noisy)
-        data = torch.from_numpy(data)
-        noisy = torch.from_numpy(noisy)
-        noisy = noisy.unsqueeze(0)
-        noisy = noisy.unsqueeze(1)
-        output = model(noisy.float()).squeeze(0).squeeze(0).detach().numpy()
+        flipx, flipy, rot = get_flips()
+        dataSharp = get_bin_weights(branchSharp, image, flipx, flipy, rot).copy()
+        np.savetxt(args.outf+'/samples/sharp' + str(image) + '.txt', dataSharp)
+        dataFuzz = get_bin_weights(branchFuzz, image, flipx, flipy, rot).copy()
+        np.savetxt(args.outf+'/samples/fuzzy' + str(image) + '.txt', dataFuzz)
+        dataSharp = torch.from_numpy(dataSharp)
+        dataFuzz = torch.from_numpy(dataFuzz)
+        dataFuzz = dataFuzz.unsqueeze(0)
+        dataFuzz = dataFuzz.unsqueeze(1)
+        output = model(dataFuzz.float()).squeeze(0).squeeze(0).detach().numpy()
         np.savetxt(args.outf+'/samples/output' + str(image) + '.txt', output)
-        truth = data.numpy()
-        noisy = noisy.numpy()
-        diff = output-truth
-        noisy_diff = noisy-truth
+        sharp = dataSharp.numpy()
+        fuzzy = dataFuzz.numpy()
+        diff = output-sharp
+        fuzzy_diff = fuzzy-sharp
         np.savetxt(args.outf+'/samples/diff' + str(image) + '.txt', diff)
-        del data
-        del noisy
+        del sharp
+        del fuzzy
         del output
         del diff
     model.to('cuda')
 
+
+#gets data needed to make histograms of sample data
+#assumes that the xmin, xmax, ymin, and ymax values are the same 
+#for all four high and low quality training and validation files
+file = up.open(args.valfileSharp)
+tree = file["g4SimHits/tree"]
+x_min = tree["xmin"].array().to_numpy()[0]
+x_max = tree["xmax"].array().to_numpy()[0]
+y_min = tree["ymin"].array().to_numpy()[0]
+y_max = tree["ymax"].array().to_numpy()[0]
+x_bins = tree["xbins"].array().to_numpy()[0]
+y_bins = tree["ybins"].array().to_numpy()[0]
+
+#makes histograms given bin weights listed in .txt file    
+def make_plots(fin, data):
+    binweights = np.loadtxt(fin)
+    binarray = []
+    for i, elem in enumerate(binweights):
+        for j, elem in enumerate(binweights[i]):
+            binarray.append(binweights[i][j])
+            
+    if 'output' in fin:
+        event_number = fin.replace(args.outf+'/samples/'+'output','')
+        event_number = event_number.replace('.txt','')
+        event_number = int(event_number)
+        means = np.mean(data[0][0].numpy())
+        stdevs = np.std(data[0][0].numpy())
+        
+        for i, elem in enumerate(binarray):
+            binarray[i] *= stdevs
+            binarray[i] += means
+        
+    else:
+        print('not output') 
+        
+    #builds axes for histogram given min/max and bin number
+    x_axis = []
+    count = 0
+    x_start = x_min
+    while count < 100:
```

Comment:
should these 100s be x_bins and y_bins?

Reply:
```suggestion
        x_axis = []
    count = 0
    x_start = x_min
    while count < y_bins:
        for i in range(x_bins):
            x = x_start + ((x_max-x_min)/x_bins)*float(i)
            x_axis.append(x)
        count = count + 1

    y_axis = []
    y_start = y_min
    for i in range(y_bins):
        y = y_start + ((y_max-y_min)/y_bins)*float(i)
        count = 0
        while count < x_bins:
            y_axis.append(y)
            count = count + 1
```

yes definitely. Also I know this is simple but I'm getting a little turned around with where each should go... also there's something similar going on  here: https://github.com/kpedro88/SimDenoising_training/blob/00b63b75282fb7c840e75c9addba6c4a24768046/dataset.py#L28-L35
so i should probably make a function so that doesn't get mixed up... just let me know if there's any reason that needs to stay separate